ekncontent/ekncontent/eknc-content-object-model.c

```diff
@@ -0,0 +1,578 @@
+      break;
+
+    case PROP_LAST_MODIFIED_DATE:
+      g_free (priv->last_modified_date);
+      priv->last_modified_date = g_value_dup_string (value);
+      break;
+
+    case PROP_LICENSE:
+      g_free (priv->license);
+      priv->license = g_value_dup_string (value);
+      break;
+
+    case PROP_FEATURED:
+      priv->featured = g_value_get_boolean (value);
+      break;
+
+    case PROP_TAGS:
+      priv->tags = g_value_dup_variant (value);
+      break;
+
+    case PROP_RESOURCES:
+      priv->resources = g_value_dup_variant (value);
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+    }
+}
+
+static void
+eknc_content_object_model_constructed (GObject *object)
+{
+  EkncContentObjectModel *self = EKNC_CONTENT_OBJECT_MODEL (object);
+  EkncContentObjectModelPrivate *priv = eknc_content_object_model_get_instance_private (self);
+
+  if (g_strcmp0 (priv->ekn_id, "") == 0)
+    {
+      // Note: This is only for ensuring the invariant of "each model has an
+      // EKN ID" in tests. It is illegal to create a model in production code
+      // with no EKN ID.
+      g_autofree gchar *sum = NULL;
+      static guint64 MOCKED_OBJECTS = 0;
+      sum = g_compute_checksum_for_data (G_CHECKSUM_SHA1,
+                                         (guchar *) &MOCKED_OBJECTS,
+                                         sizeof (MOCKED_OBJECTS));
+      MOCKED_OBJECTS++;
+      priv->ekn_id = g_strconcat ("ekn:///", sum, NULL);
+    }
+}
+
+static void
+eknc_content_object_model_finalize (GObject *object)
+{
+  EkncContentObjectModel *self = EKNC_CONTENT_OBJECT_MODEL (object);
+  EkncContentObjectModelPrivate *priv = eknc_content_object_model_get_instance_private (self);
+
+  g_free (priv->ekn_id);
+  g_free (priv->title);
+  g_free (priv->original_title);
+  g_free (priv->original_uri);
+  g_free (priv->thumbnail_uri);
+  g_free (priv->language);
+  g_free (priv->copyright_holder);
+  g_free (priv->source_uri);
+  g_free (priv->content_type);
+  g_free (priv->synopsis);
+  g_free (priv->last_modified_date);
+  g_free (priv->license);
+  g_clear_pointer (&priv->tags, g_variant_unref);
+  g_clear_pointer (&priv->resources, g_variant_unref);
+}
+
+static void
+eknc_content_object_model_class_init (EkncContentObjectModelClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->get_property = eknc_content_object_model_get_property;
+  object_class->set_property = eknc_content_object_model_set_property;
+  object_class->constructed = eknc_content_object_model_constructed;
+  object_class->finalize = eknc_content_object_model_finalize;
+
+  /**
+   * EkncContentObjectModel:ekn-id:
+   *
+   * Unique ID of the model
+   *
+   * This is an internal ID assigned by EKN.
+   * If none is provided, the model will generate its own id.
+   */
+  eknc_content_object_model_props[PROP_EKN_ID] =
+    g_param_spec_string ("ekn-id", "Object's ID",
```

Comment:
Sure. Commit wise I would do this as a new thing tagged on the end. It'll make this branch easier for me to drive/reason about. But I can add that on when I get through the bulk of the port.

As to renaming, it does not seem super clear cut. `Id` would be more consistent with other libraries, shard included. 'ekn-id' is a name that ends up sticking in our heads and discussions though. You can say 'ekn-id' and everyone knows what you are talking about without needed to contextualize. Though maybe not a outsider friendly as a term.